### ISSUE TITLE
UPDATE...RETURNING causes exception if no results

### ISSUE THREAD

**Issue by jeremyrsmith** [finagle/finagle-postgres#8]
You must call client.select() in order to get results out of an UPDATE...RETURNING query.  But, a CommandComplete tag for Update(), in the absence of a row buffer, leads to an OK() response.  Since the select() method treats this as an error condition, you must be very sure that there will be at least one row updated, or the UPDATE...RETURNING query will lead to a semantically meaningless exception being thrown.

select()'s behavior could be changed to silently drop the OK() message, since it is not typically indicative of an error.  No information would be lost, because in the case where an OK() message was received by select()'s callback, no rows were updated so the parameter from the Update() tag is always zero.  Instead, the OK(_) case could just map an empty row set:

``` scala
case OK(_) => Seq.empty[Row].map(f)
```

I understand why the correlating error is thrown from exec(), because in that case rows were received and there is no mapper supplied to map them.  That is more easily considered an error condition.

Reproduce:

``` scala
for {
_ <- client.exec("CREATE TABLE update_returning_test (foo integer)")
r <- client.query("UPDATE update_returning_test SET foo = $1 returning foo", 5)(row => row.get[Int]("foo"))
} yield r
```

**Comment by jeremyrsmith** [comment]
Cause is here:
https://github.com/finagle/finagle-postgres/blob/master/src/main/scala/com/twitter/finagle/postgres/Client.scala#L226

**Comment by vkostyukov** [comment]
Fixed by #10.
